src/state_machine.zig

```diff
@@ -576,6 +577,38 @@ pub fn StateMachineType(
             callback(self);
         }
 
+        fn scope_open(self: *StateMachine, operation: Operation) void {
+            assert(operation != .lookup_accounts and operation != .lookup_transfers);
```

Comment:
I think this assertion is probably unnecessary/redundant thanks to the following `switch`/`unreachable`. But if you think it is clearer to include, then I would split the conjunction into two assertions:

```suggestion
            assert(operation != .lookup_accounts);
            assert(operation != .lookup_transfers);
```

(Likewise for `scope_close()`).

Reply:
I borrowed it from `.execute()` - agreed doesn't make sense with the `switch`.